hoot-core/src/main/cpp/hoot/core/io/HootApiDbWriter.cpp

```diff
@@ -428,6 +430,10 @@ void HootApiDbWriter::writePartial(const ConstNodePtr& n)
     {
       _hootdb.updateNode(nodeId, n->getY(), n->getX(), n->getVersion() + 1, tags);
```

Comment:
Everything looks good.  I had a question about this like here where we are incrementing the version by one ourselves.  Will this cause an issue in the upload?  Maybe because we've conflated the node/way/relation and we are changing the version.  This would cause `changeset-apply` to have to communicate with the server and get the current version and go back one in order for it to be successful.  I'm not sure, just a thought I had.  Maybe we only increment the version when we aren't doing a grail conflation.

Reply:
You could be right about that. I only focused on inserts for these changes since that was all that requested and left the update behavior unchanged. Not sure what will happen when that changes.

Reply:
I could add another option for that, though. Or tie it to the other option and rename it.

Reply:
was thinking in the changeset derive we would want to use the ref version when diff'ing for this reason

Reply:
because we will also increment the version if users edit the data in the UI to merge or otherwise resolve reviews or maybe even tweak the data with edit tools.  Hoot services will increment the version for that edit.

Reply:
Ok, that makes sense.  The reference dataset should have the current `version` number in the element while the secondary dataset could potentially have a bunch of changes made, conflation, hand editing, reviews, etc. and we don't want to use that version number because that will cause a conflict that has to be resolved.

Reply:
Making this change in #3870